src/Oscoin/Storage/Block/Abstract.hs

```diff
@@ -7,37 +7,53 @@
 -}
 
 module Oscoin.Storage.Block.Abstract
-    ( BlockStore(..)
+    ( BlockStoreReader(..)
+    , BlockStore
 
-    , hoistBlockStore
-    , noValidation
+    , hoistBlockStoreReader
     , insertBlocksNaive
     , isNovelBlock
+
+    -- * Internals
+    , BlockStoreWriter(..)
+    , hoistBlockStoreWriter
     ) where
 
 import           Oscoin.Prelude
 
-import           Oscoin.Consensus.Types (Validate)
 import           Oscoin.Crypto.Blockchain (TxLookup)
 import           Oscoin.Crypto.Blockchain.Block
 import           Oscoin.Crypto.Hash (Hashed)
+import           Oscoin.Time.Chrono (NewestFirst, OldestFirst)
+
+-- The full 'BlockStore' API is the composition of its public and private parts.
+type BlockStore c tx s m =
+    ( BlockStoreReader c tx s m
+    , BlockStoreWriter c tx s m
+    )
 
--- | An handle over a block storage backend.
-data BlockStore c tx s m = BlockStore
-    { scoreBlock      :: Block c tx (Sealed c s) -> Score
-    -- ^ When given a 'Block', return its 'Score'.
-    , insertBlock     :: Block c tx (Sealed c s) -> m ()
+-- | The internals for this 'BlockStore', i.e. the private API which shouldn't
+-- be visible to \"read-only consumers\".
+data BlockStoreWriter c tx s m = BlockStoreWriter
+    { insertBlock     :: Block c tx (Sealed c s) -> m ()
```

Comment:
It would be great if we could document that this assumes that the parent of `block` the same is as the result of `getTip`.

Reply:
3d20d7a